tests/tests.py

```diff
@@ -366,3 +374,107 @@ def test_expiry_is_present(self):
             response.data['expiry'],
             AuthToken.objects.first().expiry
         )
+
+    def test_token_expiry_is_extended_via_refresh_endpoint(self):
+
+        ttl = knox_settings.TOKEN_TTL
+        original_time = datetime(2018, 7, 25, 0, 0, 0, 0)
+
+        with freeze_time(original_time):
+            instance, token = AuthToken.objects.create(user=self.user)
+
+        self.client.credentials(HTTP_AUTHORIZATION=('Token %s' % token))
+        five_hours_later = original_time + timedelta(hours=5)
+
+        with override_settings(REST_KNOX=refresh_endpoint_knox):
+            reload_module(auth)  # necessary to reload settings in core code
+            reload_module(views)
+            root_url = reverse('api-root')
+            url = reverse('knox_refresh')
+            with freeze_time(five_hours_later):
+                response = self.client.post(
+                    url,
+                    {},
+                    format='json'
+                )
+        reload_module(views)
+        reload_module(auth)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.data), 1)
+        self.assertIn(
+            'expiry',
+            response.data
+        )
+        expected_expiry = original_time + ttl + timedelta(hours=5)
+        self.assertEqual(
+            make_naive(response.data['expiry']),
+            expected_expiry
+        )
+
+        # original expiry date was extended:
+        new_expiry = AuthToken.objects.get().expiry
+        self.assertEqual(
+            new_expiry.replace(tzinfo=None),
+            expected_expiry,
+            "Expiry time should have been extended to {} but is {}.".format(
+                expected_expiry,
+                new_expiry
+            )
+        )
+
+        # token works after original expiry:
+        after_original_expiry = original_time + ttl + timedelta(hours=1)
+        with freeze_time(after_original_expiry):
+            response = self.client.get(root_url, {}, format='json')
+            self.assertEqual(response.status_code, 200)
+
+        # token does not work after new expiry:
+        new_expiry = AuthToken.objects.get().expiry
+        with freeze_time(new_expiry + timedelta(seconds=1)):
+            response = self.client.get(root_url, {}, format='json')
+            self.assertEqual(response.status_code, 401)
+
+    def test_endpoint_not_enabled_403_forbidden(self):
+
```

Comment:
Add a comment that by default the settings is false and the endpoint is thus disabled.